### PR TITLE
resolves #80, with fixes for clippy warnings, and a nix flake for the nix package manager. 

### LINKED ISSUE #80: graceful way to end `inputbot::handle_input_events`
Hi I'm building a gui around this crate, and I enjoy using it. Currently I start `handle_input_events` in a new thread so my gui doesn't freeze, and there's no correct way to end this loop. I could kill the thread but that seems wrong. I'd be open to submitting a pr if i could get some information on what is needed since every platform has a different impl of `handle_input_events`.

### PR DESCRIPTION
**Pull request by Redhawk18**

### CODE REVIEW

**Comment by Redhawk18** [comment]
also, should i worry about how this compiles on windows? its more work but i have a windows machine if i need to add a impl for this function on window.

**Comment by Redhawk18** [comment]
So after pulling the project up on my windows machine, it seems that the event loop on the windows side isn't handled by this crate so the solution on linux can't be implemented what I see.

File: src/linux/mod.rs [hunk 1]
```diff
@@ -33,11 +39,8 @@ type KeyStatesMap = HashMap<KeybdKey, bool>;
 
 static BUTTON_STATES: Lazy<Mutex<ButtonStatesMap>> =
     Lazy::new(|| Mutex::new(ButtonStatesMap::new()));
+static END_HANDLE_EVENTS: AtomicBool = AtomicBool::new(false);
```
Comment (obv-mikhail) [inline src/linux/mod.rs]: Maybe rename this to be positive? `HANDLE_EVENTS`?

Comment (Redhawk18) [reply src/linux/mod.rs]: okay, but what about the problem on the windows side?

**Comment by obv-mikhail** [comment]
Don't worry about Windows.

**Comment by Redhawk18** [comment]
> Don't worry about Windows.

why not?

**Comment by obv-mikhail** [comment]
> > Don't worry about Windows.
> 
> why not?

Does it still build? I think there would simply be a small API difference, so this can could be addressed later as needed.

**Comment by Redhawk18** [comment]
sounds good, made my changes

**Review by obv-mikhail — APPROVED** [review]

**Pull request merged**